packages/flutter/lib/src/http/mojo_client.dart

```diff
@@ -130,28 +130,17 @@ class MojoClient {
   ///
   /// The Future will emit a [ClientException] if the response doesn't have a
   /// success status code.
-  Future<mojo.MojoDataPipeConsumer> readDataPipe(dynamic url, { Map<String, String> headers }) async {
+  Future<mojo.MojoDataPipeConsumer> readDataPipe(dynamic url, { Map<String, String> headers }) {
+    Completer<mojo.MojoDataPipeConsumer> completer = new Completer<mojo.MojoDataPipeConsumer>();
     mojom.UrlLoaderProxy loader = new mojom.UrlLoaderProxy.unbound();
-    mojom.UrlRequest request = _prepareRequest('GET', url, headers);
-    mojom.UrlResponse response;
-    try {
-      networkService.createUrlLoader(loader);
-      response = (await loader.start(request)).response;
-    } catch (exception, stack) {
-      FlutterError.reportError(new FlutterErrorDetails(
-        exception: exception,
-        stack: stack,
-        library: 'networking HTTP library',
-        context: 'while sending bytes to the Mojo network library',
-        silent: true
-      ));
-      return null;
-    } finally {
+    networkService.createUrlLoader(loader);
+    loader.start(_prepareRequest('GET', url, headers), (mojom.UrlResponse response) {
       loader.close();
-    }
-    if (response.statusCode < 400)
-      return response.body;
-    throw new Exception("Request to $url failed with status ${response.statusCode}.");
+      if (response.statusCode < 400)
+        completer.complete(response.body);
+      completer.completeError(new Exception("Request to $url failed with status ${response.statusCode}."));
```

Comment:
we should still report the error locally with silent:true, so that during development you can see 404s and so forth in the logs (silent:true means nothing is logged in release builds)